core/src/main/java/org/opensearch/remote/metadata/common/SdkClientUtils.java

```diff
@@ -9,25 +9,178 @@
+import org.opensearch.action.get.GetResponse;
+import org.opensearch.action.index.IndexResponse;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.update.UpdateResponse;
 import org.opensearch.common.action.ActionFuture;
 import org.opensearch.common.util.concurrent.FutureUtils;
 import org.opensearch.common.util.concurrent.UncategorizedExecutionException;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.remote.metadata.client.BulkDataObjectResponse;
+import org.opensearch.remote.metadata.client.DeleteDataObjectResponse;
+import org.opensearch.remote.metadata.client.GetDataObjectResponse;
+import org.opensearch.remote.metadata.client.PutDataObjectResponse;
+import org.opensearch.remote.metadata.client.SearchDataObjectResponse;
+import org.opensearch.remote.metadata.client.UpdateDataObjectResponse;
 
+import java.io.IOException;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Locale;
 import java.util.concurrent.CompletionException;
 import java.util.concurrent.ExecutionException;
+import java.util.function.BiConsumer;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
+import static org.opensearch.core.rest.RestStatus.INTERNAL_SERVER_ERROR;
+
 /**
  * Utility methods for client implementations
  */
 public class SdkClientUtils {
 
     private SdkClientUtils() {}
 
+    /**
+     * Wraps the completion of a PUT operation from the SdkClient into a format compatible with an ActionListener.
+     *
+     * @param listener The ActionListener that will receive the parsed IndexResponse or any errors
+     * @return A BiConsumer that can be used directly with CompletionStage's whenComplete method
+     */
+    public static BiConsumer<PutDataObjectResponse, Throwable> wrapPutCompletion(ActionListener<IndexResponse> listener) {
+        return (r, throwable) -> {
+            if (throwable == null) {
+                try {
+                    IndexResponse indexResponse = IndexResponse.fromXContent(r.parser());
+                    listener.onResponse(indexResponse);
+                } catch (IOException e) {
+                    handleParseFailure(listener, "put");
+                }
+            } else {
+                handleThrowable(listener, throwable);
+            }
+        };
+    }
+
+    /**
+     * Wraps the completion of a GET operation from the SdkClient into a format compatible with an ActionListener.
+     *
+     * @param listener The ActionListener that will receive the parsed GetResponse or any errors
+     * @return A BiConsumer that can be used directly with CompletionStage's whenComplete method
+     */
+    public static BiConsumer<GetDataObjectResponse, Throwable> wrapGetCompletion(ActionListener<GetResponse> listener) {
+        return (r, throwable) -> {
+            if (throwable == null) {
+                try {
+                    GetResponse getResponse = GetResponse.fromXContent(r.parser());
```

Comment:
In our end we used to do like this:
```
GetResponse gr = getDataObjectResponse.parser() == null ? null : GetResponse.fromXContent(getDataObjectResponse.parser());
```

Shouldn't we check null for `r.parser()`?

Reply:
Good point, especially since we only catch IOException